cmdpolling/continuousreport.go

```diff
@@ -0,0 +1,176 @@
+package cmdpolling
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
+	"github.com/ingrammicro/concerto/api/polling"
+	"github.com/ingrammicro/concerto/api/types"
+	"github.com/ingrammicro/concerto/cmd"
+	"github.com/ingrammicro/concerto/utils"
+	"github.com/ingrammicro/concerto/utils/format"
+)
+
+const (
+	DefaultThresholdLines = 10
+	DefaultThresholdTime  = 10
+	DefaultThresholdBytes = 500
+	RetriesNumber         = 5
+	RetriesFactor         = 3
+)
+
+type Threshold struct {
+	lines int
+	time  int
+	bytes int
+}
+
+func cmdContinuousReportRun(c *cli.Context) error {
+	log.Debug("cmdContinuousReportRun")
+
+	formatter := format.GetFormatter()
+	pollingSvc := cmd.WireUpPolling(c)
+
+	// cli command argument
+	var commandArg string
+	if c.Args().Present() {
+		commandArg = c.Args().First()
+	} else {
+		formatter.PrintFatal("argument missing", errors.New("a script or command is required"))
+	}
+
+	// cli command thresholds flags
+	thresholdLines := c.Int("lines")
+	if !(thresholdLines > 0) {
+		thresholdLines = DefaultThresholdLines
+	}
+	thresholdTime := c.Int("time")
+	if !(thresholdTime > 0) {
+		thresholdTime = DefaultThresholdTime
+	}
+	thresholdBytes := c.Int("bytes")
+	if !(thresholdBytes > 0) {
+		thresholdBytes = DefaultThresholdBytes
+	}
+	threshold := Threshold{lines: thresholdLines, time: thresholdTime, bytes: thresholdBytes}
+	log.Debug("Threshold", threshold)
+
+	if err := processCommand(pollingSvc, commandArg, threshold); err != nil {
+		formatter.PrintFatal("cannot process continuous report command", err)
+	}
+
+	log.Info("completed")
+	return nil
+}
+
+func processCommand(pollingSvc *polling.PollingService, commandArg string, threshold Threshold) error {
+	log.Debug("processCommand")
+
+	var newCmd *exec.Cmd
+	if runtime.GOOS == "windows" {
+		newCmd = exec.Command("cmd", "/C", commandArg)
+	} else {
+		newCmd = exec.Command("/bin/sh", "-c", commandArg)
```

Comment:
With respect to the integration of this with ingrammicro/tapp#722, and regarding windows machines, do you know if there is any limit to the size the /C flag value of cmd accepts? If there is, we may need to write the commandArg to a file and run it as a batch script

Reply:
Done. 
In addition, i would like to refactor this 'processCommand' method to utils/exec... 

This implies externalice this functionallity to exec file, and use funcion definition as a handler for the chunks:
fn := func(chunk string) error {
...
}

Thus, passing 'sendChunks' method as parameter to this method: "processCommand(fn func(chunk string) error, X, Y)", and in charge of process the chunks for a given command.

The idea is to make this reusable, and clean this archive...
What do youn think?

Reply:
:+1: